2023-09-18 17:43:37.877832: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2023-09-18 17:43:37.878017: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2023-09-18 17:43:45,532 | app.py:151 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2023-09-18 17:43:45,628 | app.py:172 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2023-09-18 17:43:45,628 | server.py:86 | Initializing global parameters
INFO flwr 2023-09-18 17:43:45,628 | server.py:273 | Requesting initial parameters from one random client
INFO flwr 2023-09-18 17:43:45,637 | server.py:277 | Received initial parameters from one random client
INFO flwr 2023-09-18 17:43:45,637 | server.py:88 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:43:45,654 | server.py:95 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.39666666666666667}
INFO flwr 2023-09-18 17:43:45,655 | server.py:101 | FL starting
DEBUG flwr 2023-09-18 17:43:45,858 | server.py:223 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2023-09-18 17:43:45,880 | server.py:237 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2023-09-18 17:43:45,882 | fedavg.py:243 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:43:45,890 | server.py:125 | fit progress: (1, 0.39136646412872317, {'accuracy': 0.8}, 0.23536900000000038)
DEBUG flwr 2023-09-18 17:43:45,891 | server.py:173 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:43:45,914 | server.py:187 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2023-09-18 17:43:45,914 | fedavg.py:274 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2023-09-18 17:43:45,915 | server.py:223 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:43:45,933 | server.py:237 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:43:45,943 | server.py:125 | fit progress: (2, 0.39211681716147906, {'accuracy': 0.81}, 0.2883873000000001)
DEBUG flwr 2023-09-18 17:43:45,944 | server.py:173 | evaluate_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:43:45,961 | server.py:187 | evaluate_round 2 received 3 results and 0 failures
DEBUG flwr 2023-09-18 17:43:45,961 | server.py:223 | fit_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:43:45,984 | server.py:237 | fit_round 3 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:43:45,998 | server.py:125 | fit progress: (3, 0.3921189921871429, {'accuracy': 0.81}, 0.3423356999999996)
DEBUG flwr 2023-09-18 17:43:45,998 | server.py:173 | evaluate_round 3: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:43:46,017 | server.py:187 | evaluate_round 3 received 3 results and 0 failures
DEBUG flwr 2023-09-18 17:43:46,018 | server.py:223 | fit_round 4: strategy sampled 3 clients (out of 3)
DEBUG flwr 2023-09-18 17:43:46,038 | server.py:237 | fit_round 4 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:43:46,053 | server.py:125 | fit progress: (4, 0.39211906581108574, {'accuracy': 0.81}, 0.3970127000000012)
DEBUG flwr 2023-09-18 17:43:46,053 | server.py:173 | evaluate_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:43:46,077 | server.py:187 | evaluate_round 4 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:43:46,077 | server.py:223 | fit_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:43:46,106 | server.py:237 | fit_round 5 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:43:46,118 | server.py:125 | fit progress: (5, 0.39365820732930423, {'accuracy': 0.8033333333333333}, 0.463265100000001)
DEBUG flwr 2023-09-18 17:43:46,118 | server.py:173 | evaluate_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:43:46,139 | server.py:187 | evaluate_round 5 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:43:46,140 | server.py:223 | fit_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:43:46,167 | server.py:237 | fit_round 6 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:43:46,180 | server.py:125 | fit progress: (6, 0.3936440938963407, {'accuracy': 0.8033333333333333}, 0.5244002000000005)
DEBUG flwr 2023-09-18 17:43:46,180 | server.py:173 | evaluate_round 6: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:43:46,201 | server.py:187 | evaluate_round 6 received 5 results and 0 failures
DEBUG flwr 2023-09-18 17:43:46,202 | server.py:223 | fit_round 7: strategy sampled 5 clients (out of 5)
DEBUG flwr 2023-09-18 17:43:46,226 | server.py:237 | fit_round 7 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:43:46,238 | server.py:125 | fit progress: (7, 0.3936450514755527, {'accuracy': 0.8033333333333333}, 0.5822415000000003)
DEBUG flwr 2023-09-18 17:43:46,238 | server.py:173 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:43:46,263 | server.py:187 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:43:46,263 | server.py:223 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:43:46,298 | server.py:237 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:43:46,312 | server.py:125 | fit progress: (8, 0.3920130697053319, {'accuracy': 0.8066666666666666}, 0.6565182000000007)
DEBUG flwr 2023-09-18 17:43:46,312 | server.py:173 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:43:46,336 | server.py:187 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:43:46,337 | server.py:223 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:43:46,368 | server.py:237 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:43:46,382 | server.py:125 | fit progress: (9, 0.39202016207906204, {'accuracy': 0.8066666666666666}, 0.7273585000000011)
DEBUG flwr 2023-09-18 17:43:46,383 | server.py:173 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:43:46,410 | server.py:187 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2023-09-18 17:43:46,410 | server.py:223 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:43:46,450 | server.py:237 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2023-09-18 17:43:46,463 | server.py:125 | fit progress: (10, 0.39201942589024136, {'accuracy': 0.8066666666666666}, 0.8076465000000006)
DEBUG flwr 2023-09-18 17:43:46,463 | server.py:173 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2023-09-18 17:43:46,490 | server.py:187 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2023-09-18 17:43:46,491 | server.py:147 | FL finished in 0.8353426000000006
INFO flwr 2023-09-18 17:43:46,491 | app.py:218 | app_fit: losses_distributed [(1, 0.39486186392605305), (2, 0.39334152080118656), (3, 0.3933422211557627), (4, 0.40444858301253545), (5, 0.40338897988909767), (6, 0.40338450812158133), (7, 0.40691963391644614), (8, 0.4065522675003324), (9, 0.40655333484922135), (10, 0.4065534093550273)]
INFO flwr 2023-09-18 17:43:46,491 | app.py:219 | app_fit: metrics_distributed_fit {}
INFO flwr 2023-09-18 17:43:46,491 | app.py:220 | app_fit: metrics_distributed {}
INFO flwr 2023-09-18 17:43:46,491 | app.py:221 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.39136646412872317), (2, 0.39211681716147906), (3, 0.3921189921871429), (4, 0.39211906581108574), (5, 0.39365820732930423), (6, 0.3936440938963407), (7, 0.3936450514755527), (8, 0.3920130697053319), (9, 0.39202016207906204), (10, 0.39201942589024136)]
INFO flwr 2023-09-18 17:43:46,492 | app.py:222 | app_fit: metrics_centralized {'accuracy': [(0, 0.39666666666666667), (1, 0.8), (2, 0.81), (3, 0.81), (4, 0.81), (5, 0.8033333333333333), (6, 0.8033333333333333), (7, 0.8033333333333333), (8, 0.8066666666666666), (9, 0.8066666666666666), (10, 0.8066666666666666)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.0996612 ,  0.96373856,  0.71296191,  0.4023511 ,  0.20552414,
        -0.01111928,  0.03633247, -0.13627745]]), array([0.49174085])]
[array([[-2.0996612 ,  0.96373856,  0.71296191,  0.4023511 ,  0.20552414,
        -0.01111928,  0.03633247, -0.13627745]]), array([0.49174085])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03061576,  0.98425906,  0.76061625,  0.34626643,  0.25494102,
         0.0098756 ,  0.00698818, -0.02139523]]), array([0.24066445])]
[array([[-2.03061576,  0.98425906,  0.76061625,  0.34626643,  0.25494102,
         0.0098756 ,  0.00698818, -0.02139523]]), array([0.24066445])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.0303623 ,  0.98412501,  0.76053822,  0.34623056,  0.2550012 ,
         0.00992538,  0.0069312 , -0.02136712]]), array([0.24043534])]
[array([[-2.0303623 ,  0.98412501,  0.76053822,  0.34623056,  0.2550012 ,
         0.00992538,  0.0069312 , -0.02136712]]), array([0.24043534])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.03033345,  0.98412042,  0.76054742,  0.34624188,  0.25500939,
         0.00994736,  0.00693159, -0.0213622 ]]), array([0.24034514])]
[array([[-2.03033345,  0.98412042,  0.76054742,  0.34624188,  0.25500939,
         0.00994736,  0.00693159, -0.0213622 ]]), array([0.24034514])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.97411388,  1.01349794,  0.77866157,  0.38354568,  0.19156729,
         0.06996972, -0.02708902, -0.02834142]]), array([0.08082487])]
[array([[-1.97411388,  1.01349794,  0.77866157,  0.38354568,  0.19156729,
         0.06996972, -0.02708902, -0.02834142]]), array([0.08082487])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.97666075,  1.01501599,  0.78027278,  0.38399183,  0.19188145,
         0.07030912, -0.02673913, -0.02858295]]), array([0.07966716])]
[array([[-1.97666075,  1.01501599,  0.78027278,  0.38399183,  0.19188145,
         0.07030912, -0.02673913, -0.02858295]]), array([0.07966716])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.97645626,  1.01492105,  0.78018639,  0.38392494,  0.19180076,
         0.07029789, -0.02669353, -0.02859711]]), array([0.07974833])]
[array([[-1.97645626,  1.01492105,  0.78018639,  0.38392494,  0.19180076,
         0.07029789, -0.02669353, -0.02859711]]), array([0.07974833])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.99079117,  0.98021166,  0.80058054,  0.38233337,  0.17005755,
         0.1025759 , -0.02393239, -0.00717979]]), array([0.11749105])]
[array([[-1.99079117,  0.98021166,  0.80058054,  0.38233337,  0.17005755,
         0.1025759 , -0.02393239, -0.00717979]]), array([0.11749105])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98974572,  0.97953939,  0.80029589,  0.38217863,  0.16964838,
         0.10277873, -0.02436288, -0.00682925]]), array([0.11780661])]
[array([[-1.98974572,  0.97953939,  0.80029589,  0.38217863,  0.16964838,
         0.10277873, -0.02436288, -0.00682925]]), array([0.11780661])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-1.98966266,  0.97947882,  0.80027873,  0.38217968,  0.16962569,
         0.10276404, -0.02431737, -0.0068083 ]]), array([0.11776629])]
[array([[-1.98966266,  0.97947882,  0.80027873,  0.38217968,  0.16962569,
         0.10276404, -0.02431737, -0.0068083 ]]), array([0.11776629])]
>>>>>>>>>>>>>run time:
0.609375s
